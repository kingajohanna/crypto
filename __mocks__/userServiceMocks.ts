export const adduser = {
    status: "success",
    message: "user added successfully",
    data: {
        id: "DCzgasdfadKdDpkAWqHEw",
        email: "test@gmail.com",
        createdAt: 1648314794066,
        name: "bela",
        photoURL: "https://lh3.googleusercontent.com/a-/AOh14GjVLepb70SMpFXeAyqty8h58J9TQY7lw8wwsx4Oxg=s96-d",
    },
};

export const removeuser = {
    status: "success",
    message: "user removed successfully",
};

export const addfav = {
    status: "success",
    message: "coin added to favs successfully",
};

export const removefav = {
    status: "success",
    message: "coin removed from favs successfully",
};

export const addcoin = {
    status: "success",
    message: "coin added successfully",
    data: {
        id: "QjuLdgDdM1CkMnEMZOZg",
        userId: "DCzgasdfadKdDpkAWqH",
        coinId: "storj",
        holdings: 5,
        price: 2000,
        currency: "huf",
    },
};

export const getcoins = {
    status: "success",
    message: "fetch owned coins",
    data: [
        {
            userId: "DCzgasdfadKdDpkAWqH",
            id: "7s9J5XTfdaOthk0QIbMP",
            currency: "huf",
            coinId: "storj",
            holdings: 5,
            price: 2000,
            symbol: "storj",
            image: "https://assets.coingecko.com/coins/images/949/large/storj.png?1547034811",
            current_price: 375.7,
            name: "Storj",
        },
        {
            userId: "DCzgasdfadKdDpkAWqH",
            currency: "huf",
            holdings: 5,
            price: 2000,
            id: "QjuLdgDdM1CkMnEMZOZg",
            coinId: "storj",
            symbol: "storj",
            image: "https://assets.coingecko.com/coins/images/949/large/storj.png?1547034811",
            current_price: 375.7,
            name: "Storj",
        },
        {
            price: 159492,
            coinId: "storj",
            userId: "DCzgasdfadKdDpkAWqH",
            currency: "huf",
            id: "VB8cBlj3fRJ7cPSRiqFg",
            holdings: 9,
            symbol: "storj",
            image: "https://assets.coingecko.com/coins/images/949/large/storj.png?1547034811",
            current_price: 375.7,
            name: "Storj",
        },
        {
            id: "ZlH7U70fK7ChTCTogZfz",
            holdings: 5,
            currency: "huf",
            price: 2000,
            userId: "DCzgasdfadKdDpkAWqH",
            coinId: "storj",
            symbol: "storj",
            image: "https://assets.coingecko.com/coins/images/949/large/storj.png?1547034811",
            current_price: 375.7,
            name: "Storj",
        },
        {
            coinId: "storj",
            holdings: 5,
            price: 2000,
            id: "hssyU4lUp7R53eztlUsA",
            userId: "DCzgasdfadKdDpkAWqH",
            currency: "huf",
            symbol: "storj",
            image: "https://assets.coingecko.com/coins/images/949/large/storj.png?1547034811",
            current_price: 375.7,
            name: "Storj",
        },
    ],
};

export const setcoin = {
    status: "success",
    message: "coin edited successfully",
};
