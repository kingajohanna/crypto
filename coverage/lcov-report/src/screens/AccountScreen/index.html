
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/screens/AccountScreen</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> src/screens/AccountScreen</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">47.22% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>34/72</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">0% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/22</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">10.52% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>2/19</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">47.76% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>32/67</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="UserLoggedIn.tsx"><a href="UserLoggedIn.tsx.html">UserLoggedIn.tsx</a></td>
	<td data-value="80" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 80%"></div><div class="cover-empty" style="width: 20%"></div></div>
	</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="10" class="abs high">8/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	<td data-value="77.77" class="pct medium">77.77%</td>
	<td data-value="9" class="abs medium">7/9</td>
	</tr>

<tr>
	<td class="file low" data-value="UserNotLoggedIn.tsx"><a href="UserNotLoggedIn.tsx.html">UserNotLoggedIn.tsx</a></td>
	<td data-value="41.93" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 41%"></div><div class="cover-empty" style="width: 59%"></div></div>
	</td>
	<td data-value="41.93" class="pct low">41.93%</td>
	<td data-value="62" class="abs low">26/62</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="22" class="abs low">0/22</td>
	<td data-value="6.25" class="pct low">6.25%</td>
	<td data-value="16" class="abs low">1/16</td>
	<td data-value="43.1" class="pct low">43.1%</td>
	<td data-value="58" class="abs low">25/58</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Thu Apr 28 2022 12:48:39 GMT+0200 (GMT+02:00)
            </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    