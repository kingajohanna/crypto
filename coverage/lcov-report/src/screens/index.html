
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/screens</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/screens</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">72.28% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>60/83</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">55% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>11/20</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">40.9% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>9/22</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">68.91% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>51/74</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="AccountScreen.tsx"><a href="AccountScreen.tsx.html">AccountScreen.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	</tr>

<tr>
	<td class="file high" data-value="FavouritesScreen.tsx"><a href="FavouritesScreen.tsx.html">FavouritesScreen.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	</tr>

<tr>
	<td class="file medium" data-value="MarketScreen.tsx"><a href="MarketScreen.tsx.html">MarketScreen.tsx</a></td>
	<td data-value="54" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 54%"></div><div class="cover-empty" style="width: 46%"></div></div>
	</td>
	<td data-value="54" class="pct medium">54%</td>
	<td data-value="50" class="abs medium">27/50</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="8" class="abs low">2/8</td>
	<td data-value="18.75" class="pct low">18.75%</td>
	<td data-value="16" class="abs low">3/16</td>
	<td data-value="51.06" class="pct medium">51.06%</td>
	<td data-value="47" class="abs medium">24/47</td>
	</tr>

<tr>
	<td class="file high" data-value="MyCoinsScreen.tsx"><a href="MyCoinsScreen.tsx.html">MyCoinsScreen.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Thu Apr 28 2022 12:48:39 GMT+0200 (GMT+02:00)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    