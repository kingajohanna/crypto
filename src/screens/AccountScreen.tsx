import React from "react";

import { Tabs } from "@constants/Tabs";

import { AccountButton } from "@components/AccountButton";
import { ScreenBackground } from "@components/ScreenBackground";

import auth from "@react-native-firebase/auth";
import { GoogleSignin } from "@react-native-google-signin/google-signin";

GoogleSignin.configure({
    webClientId: "846343653708-2vjl1i736qqn5lgfqia80g2ieqsm3ktj.apps.googleusercontent.com",
});

async function onGoogleButtonPress() {
    // Get the users ID token
    const { idToken } = await GoogleSignin.signIn();

    // Create a Google credential with the token
    const googleCredential = auth.GoogleAuthProvider.credential(idToken);

    // Sign-in the user with the credential
    const user = auth().signInWithCredential(googleCredential);

    user.then((re) => {
        console.log(re);
    });

    return user;
}

async function OnEmail() {
    return auth()
        .signInAnonymously()
        .then(() => {
            console.log("User signed in anonymously");
        })
        .catch((error) => {
            if (error.code === "auth/operation-not-allowed") {
                console.log("Enable anonymous in your firebase console.");
            }

            console.error(error);
        });
}

export const AccountScreen = () => {
    return (
        <ScreenBackground title={Tabs.account}>
            <AccountButton.Email onPress={() => OnEmail()} />
            <AccountButton.Apple />
            <AccountButton.Facebook />
            <AccountButton.Google onPress={() => onGoogleButtonPress().then(() => console.log("Signed in with Google!"))} />
            <AccountButton.Logout />
            <AccountButton.Delete />
        </ScreenBackground>
    );
};
