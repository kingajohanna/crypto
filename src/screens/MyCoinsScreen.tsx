import * as React from "react";
import { Text, View, StyleSheet } from "react-native";
import { Colors } from "@theme/Colors";
import { FAB } from "react-native-paper";
import { TabHeader } from "@components/TabHeader";
import { Tabs } from "@constants/Tabs";
import { Button } from "@components/Button";

export const MyCoinsScreen = () => {
    const styles = StyleSheet.create({
        fab: {
            position: "absolute",
            margin: 16,
            right: 4,
            bottom: 0,
        },
    });

    return (
        <View
            style={{
                flex: 1,
                alignItems: "center",
                backgroundColor: Colors.richBlack,
            }}
        >
            <TabHeader title={Tabs.mycoins} />
            <Button title="Save" />
            <Button title="Cancel" />
            <FAB style={styles.fab} icon="plus" onPress={() => {}} />
        </View>
    );
};
