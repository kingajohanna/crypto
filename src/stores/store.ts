import { authReducer, cryptoReducer, errorReducer } from "@stores/reducer";
import { combineReducers, createStore } from "redux";
import AsyncStorage from "@react-native-async-storage/async-storage";
import { persistReducer } from "redux-persist";
import persistStore from "redux-persist/es/persistStore";

const rootReducer = combineReducers({
    user: authReducer,
    crypto: cryptoReducer,
    error: errorReducer,
});

const persistConfig = {
    key: "root",
    storage: AsyncStorage,
    whitelist: ["user", "crypto"],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export type RootState = ReturnType<typeof rootReducer>;

export const store = createStore(persistedReducer);
export const persistor = persistStore(store);
