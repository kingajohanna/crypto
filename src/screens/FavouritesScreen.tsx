import React from "react";
import { shallowEqual, useSelector } from "react-redux";
import { RootState } from "@stores/store";
import { Tabs } from "@constants/Tabs";
import { ScreenBackground } from "@components/ScreenBackground";
import { UserNotLoggedIn } from "@screens/FavouritesScreen/UserNotLoggedIn";
import { UserLoggedIn } from "@screens/FavouritesScreen/UserLoggedIn";

export const FavouriteScreen = () => {
    const user = useSelector((state: RootState) => state.user, shallowEqual);

    return (
        <ScreenBackground title={Tabs.favourites}>
            {!user.isLoggedIn && <UserNotLoggedIn />}
            {user.isLoggedIn && <UserLoggedIn />}
        </ScreenBackground>
    );
};
