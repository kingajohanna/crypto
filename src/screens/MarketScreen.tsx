import React, { useEffect, useRef, useState } from "react";
import { ActivityIndicator, FlatList } from "react-native";
import RBSheet from "react-native-raw-bottom-sheet";
import { shallowEqual, useSelector } from "react-redux";
import { RootState } from "@stores/store";
import { getFavsMarket, getMarketData } from "@services/CryptoServices";
import { MarketData } from "@constants/DataTypes";
import { Tabs } from "@constants/Tabs";
import { CryptoCoin } from "@components/CryptoCoin";
import { ScreenBackground } from "@components/ScreenBackground";
import { SelectedCoinGraph } from "@components/SelectedCoinGraph";
import { Searchbar } from "@components/SearchBar";
import { ErrorComponent } from "@components/ErrorComponent";

export const MarketScreen = () => {
    const crypto = useSelector((state: RootState) => state.crypto, shallowEqual);
    const user = useSelector((state: RootState) => state.user, shallowEqual);

    const [selectedCoinData, setSelectedCoinData] = useState<MarketData | null>(null);
    const [loading, setLoading] = useState(false);
    const [data, setData] = useState(crypto.marketCoins);
    const [text, setText] = useState("");

    const refRBSheet = useRef() as React.MutableRefObject<RBSheet>;

    const fetchMarketData = async () => {
        if (user.isLoggedIn) {
            await getMarketData(user.id);
            await getFavsMarket(user.id);
        } else await getMarketData();

        setLoading(false);
    };

    const openModal = (item: MarketData) => {
        setSelectedCoinData(item);
        refRBSheet.current!.open();
    };

    const onRefresh = async () => {
        setLoading(true);
        await fetchMarketData();
    };

    useEffect(() => {
        setData(crypto.marketCoins);
    }, [crypto.marketCoins]);

    const searchData = (text: string) => {
        const filterdData = text
            ? data.filter((item: MarketData) => {
                  const itemData = item.name.toUpperCase();
                  const textData = text.toUpperCase();
                  return itemData.indexOf(textData) > -1;
              })
            : crypto.marketCoins;

        setData(filterdData);
        setText(text);
    };

    return (
        <ScreenBackground title={Tabs.market}>
            {loading && <ActivityIndicator animating size="large" style={{ paddingTop: 16 }} />}

            {crypto.marketCoins ? (
                <>
                    <Searchbar onChangeText={(text) => searchData(text)} value={text} placeholder="Search here..." />
                    <FlatList
                        keyExtractor={(item) => item.id}
                        data={data}
                        onRefresh={() => onRefresh()}
                        refreshing={loading}
                        initialNumToRender={150}
                        getItemLayout={(data, index) => ({ length: 72, offset: 72 * index, index })}
                        renderItem={({ item }) => (
                            <CryptoCoin
                                id={item.id}
                                isFavourite={item.fav}
                                onPress={() => openModal(item)}
                                name={item.name}
                                shortName={item.symbol}
                                price={item.current_price}
                                priceChange={item.price_change_percentage_7d_in_currency}
                                imageUrl={item.image}
                            />
                        )}
                    />

                    <SelectedCoinGraph selectedCoinData={selectedCoinData!} reference={refRBSheet} />
                </>
            ) : (
                <ErrorComponent onPress={() => fetchMarketData()} />
            )}
        </ScreenBackground>
    );
};
