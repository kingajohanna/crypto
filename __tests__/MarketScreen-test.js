import React from "react";
import * as renderer from "react-test-renderer";
import "jest-styled-components";
import { Provider } from "react-redux";
import { MarketScreen } from "../src/screens/MarketScreen";
import { marketdata } from "../__mocks__/cryptoServiceMocks";
import { store } from "../src/stores/store";

test("renders correctly", () => {
    // Cache original functionality
    const realUseState = React.useState;

    // Stub the initial state
    const stubInitialState = marketdata.data;

    // Mock useState before rendering your component
    jest.spyOn(React, "useState").mockImplementationOnce(() => realUseState(stubInitialState));

    const tree = renderer
        .create(
            <Provider store={store}>
                <MarketScreen />
            </Provider>,
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
