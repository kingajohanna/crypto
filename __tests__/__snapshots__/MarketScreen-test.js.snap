// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#1F2833",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#0B0C10",
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#1F2833",
            "borderBottomLeftRadius": 15,
            "borderBottomRightRadius": 15,
            "borderColor": "#1F2833",
            "borderWidth": 1,
            "height": 50,
            "justifyContent": "center",
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#C5C6C7",
              "fontSize": 24,
            },
          ]
        }
      >
        Market
      </Text>
    </View>
    <RCTScrollView
      data={Array []}
      getItemLayout={[Function]}
      initialNumToRender={100}
      keyExtractor={[Function]}
      onRefresh={[Function]}
      refreshing={false}
      renderItem={[Function]}
    >
      <View />
    </RCTScrollView>
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "portrait-upside-down",
          "landscape",
          "landscape-left",
          "landscape-right",
        ]
      }
      transparent={true}
      visible={false}
    />
  </View>
</RCTSafeAreaView>
`;
